src/discord/commands/manage.ts

```diff
@@ -0,0 +1,247 @@
+            return;
+        }
+
+        for (const id of usernameArray) {
+            if (!bookedBooking.users.includes(id)) {
+                bookedBooking.users.push(id);
+            }
+        }
+        await bookedBooking.save();
+        interaction.reply({ content: 'User(s) successfully added.', ephemeral: true });
+    }
+
+    if (manageState === 'remove') {
+        for (const id of usernameArray) {
+            if (id === bookedBooking.booker) {
+                bookError = true;
+            } else if (bookedBooking.users.includes(id)) {
+                bookedBooking.users.splice(bookedBooking.users.indexOf(id), 1);
+            }
+        }
+        await bookedBooking.save();
+
+        if (bookError && usernameArray.length > 1) {
+            interaction.reply({ content: 'User(s) successfully removed.\nPS: Cannot remove the booker. Use `/delete` to delete a booking instead.', ephemeral: true });
+        } else if (bookError) {
+            interaction.reply({ content: 'Cannot remove the booker. Use `/delete` to delete a booking instead.', ephemeral: true });
+        } else {
+            interaction.reply({ content: 'User(s) successfully removed.', ephemeral: true });
+        }
+    }
+}
+
+export default {
+    name: 'manage',
+    description: 'Manage Collaborators in Booking!',
+    options: [
+        {
+            type: 3,
+            name: 'booking-id',
+            description: 'ID of your booking',
+            required: true,
+        },
+        {
+            type: 3,
+            name: 'add-or-remove',
+            description: 'Add or remove collaborators',
+            choices: [
+                {
+                    name: 'add',
+                    value: 'add',
+                },
+                {
+                    name: 'remove',
+                    value: 'remove',
+                },
+            ],
+            required: true,
+        },
+        {
+            type: 6,
+            name: 'discord-username-1',
+            description: 'Discord username (@) of the member',
+            required: true,
+        },
+        {
+            type: 6,
+            name: 'discord-username-2',
+            description: 'Discord username (@) of the member',
+        },
+        {
+            type: 6,
+            name: 'discord-username-3',
+            description: 'Discord username (@) of the member',
+        },
+        {
+            type: 6,
+            name: 'discord-username-4',
+            description: 'Discord username (@) of the member',
+        },
+        {
+            type: 6,
+            name: 'discord-username-5',
+            description: 'Discord username (@) of the member',
+        },
+    ],
+
+    async execute(interaction: CommandInteraction | SelectMenuInteraction, userArray?: string[], maxCapacity?: number, bookingId?: string): Promise<void> {
+        if (interaction.isCommand()) {
+            await parseCommandOptions(interaction);
```

Comment:
We should rename this to something like `handleCommandInteraction()` so that the function name makes more sense. It was not immediately clear that `parseCommandOptions` also handles and responds to the interaction.